src/lxc/cgfs.c

```diff
@@ -2449,6 +2450,130 @@ static bool lxc_cgroupfs_attach(const char *name, const char *lxcpath, pid_t pid
 	return true;
 }
 
+struct chown_data {
+	const char *cgroup_path;
+	uid_t origuid;
+};
+
+/*
+ * TODO - someone should refactor this to unshare once passing all the paths
+ * to be chowned in one go
+ */
+static int chown_cgroup_wrapper(void *data)
+{
+	struct chown_data *arg = data;
+	uid_t destuid;
+	char *fpath;
+
+
+	if (setresgid(0,0,0) < 0)
+		SYSERROR("Failed to setgid to 0");
+	if (setresuid(0,0,0) < 0)
+		SYSERROR("Failed to setuid to 0");
+	if (setgroups(0, NULL) < 0)
+		SYSERROR("Failed to clear groups");
```

Comment:
On Fri, Jan 29, 2016 at 06:37:09AM -0800, Stéphane Graber wrote:
> > + * TODO - someone should refactor this to unshare once passing all the paths
> > + * to be chowned in one go
> > + */
> > +static int chown_cgroup_wrapper(void *data)
> > +{
> > +	struct chown_data *arg = data;
> > +	uid_t destuid;
> > +	char *fpath;
> > +
> > +
> > +	if (setresgid(0,0,0) < 0)
> > +		SYSERROR("Failed to setgid to 0");
> > +	if (setresuid(0,0,0) < 0)
> > +		SYSERROR("Failed to setuid to 0");
> > +	if (setgroups(0, NULL) < 0)
> > +		SYSERROR("Failed to clear groups");
> 
> Is that something that userns_exec_1 ought to do for us?

Maybe, but not all functions it calls do it, so it would
require some testing to make sure it was ok.